components/InterestZoneView/InterestZoneView.tsx

```diff
@@ -0,0 +1,148 @@
+import { InterestZone } from '../../models/zone.model';
+import DetailsForm from './InterestZoneDetailsView';
+import SterilizedCatsList from './SterilizedCatDisplay';
+import UnsterilizedCatsList from './UnsterilizedCatDisplay';
+import styles from './InterestZoneView.module.css';
+
+import Box from '@mui/material/Box';
+import Modal from '@mui/material/Modal';
+import { useCallback, useEffect, useState } from 'react';
+import Button from '@mui/material/Button';
+import Accordion from '@mui/material/Accordion';
+import AccordionSummary from '@mui/material/AccordionSummary';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import Typography from '@mui/material/Typography';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faShieldCat} from '@fortawesome/free-solid-svg-icons';
+import List from '@mui/material/List';
+import { ListItem } from '@mui/material';
+
+interface Props{
+    onClose: () => void,
+    isVisible: boolean,
+    zone: InterestZone
+}   
+
+const DetailsModal = (props: Props) => {
+    const [isEditable, setIsEditable] = useState(false);
+    const [expanded, setExpanded] = useState<string | false>(false);
+
+    const handleChange = (panel: string) => (event: React.SyntheticEvent, isExpanded: boolean) => {
+      setExpanded(isExpanded ? panel : false);
+    };
+    
+    useEffect(() => {
+        setIsEditable(false);  
+    }, [props.isVisible])
+
+
+    const handleEdit = () => {
+        setIsEditable(!isEditable);
+    };
+
+    const handleSave = () => {
+        setIsEditable(!isEditable);
+    };
+
+    const handleClose = useCallback(() => {
+        props.onClose();
+    }, []);
+
+    const handleSterilizedCatInputChange = () => {
+
+    };
+
+    const handleUnsterilizedCatInputChange = () => {
+        
+    };
+
+    const handleZoneInputChange = () => {
+        
+    };
+
+    const sterilizedCatsList = props.zone?.sterilizedCats.map((cat, cnt) => 
+        <ListItem key={cnt}>
+            <Accordion style={{width: "100%"}}>
+                <AccordionSummary expandIcon={<FontAwesomeIcon icon={faShieldCat}/>}>
+                    <Typography>
+                        Pisica nr.{cnt + 1}
+                    </Typography>
+                </AccordionSummary>
+                <AccordionDetails>
+                    <Typography>
+                        <SterilizedCatsList onChange={handleSterilizedCatInputChange} cat={cat} isEditable={isEditable}/>
+                    </Typography>
+                </AccordionDetails>
+            </Accordion>
+        </ListItem>    
```

Comment:
If not difficult, look into moving the above template into a dedicated child component.